### PR TITLE
feat(backend): Add container backend

### PR DESCRIPTION
**Pull request by kakkoyun**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -56,6 +56,7 @@ dev = [
 "virtualenv" = "isolate.backends.virtualenv:VirtualPythonEnvironment"
 "conda" = "isolate.backends.conda:CondaEnvironment"
 "local" = "isolate.backends.local:LocalPythonEnvironment"
+"container" = "isolate.backends.container:ContainerizedPythonEnvironment"
```
Comment (efiop) [inline pyproject.toml:59]: Since we are already in a container here, should we just swap `container` for `local` in controler?

Comment (kakkoyun) [reply pyproject.toml:59]: I love everyone comes to the same conclusion 🫶  However, we use the "environment" extensively over the controller plane to identify and make decisions, so swapping makes everything error-prone. I have tried. This is cleaner, and we can make further decisions on the worker nodes if we know we are in the container.

I would like to know your concerns, though. What would be the downside of having an additional backend?

**Comment by kakkoyun** [comment]
@efiop @isidentical anything else I need to care of before the merge?

**Comment by efiop** [comment]
I'm sorting out some issues with isolate right now, but if not successful shortly (like in a few hours), I'll cut a new release with your cache without my crap.

**Pull request merged**